bloom-housing/bloom
# fix: saving household member

# Pull Request Template

## Issue Overview

This PR addresses https://github.com/metrotranscom/doorway/issues/640


## Description

Previously we were not saving addresses for household members. Once we were saving, we were unable to open the edit drawer without the additional drawer state updates.

As part of this ticket, I wrote the validation piece to the public Cypress test that submits an application, so that we are checking the data is what we expect it to be on the summary view. Through the process of writing this test, I discovered we are also not saving "street2" on the primary applicant, and that if you select the non-suggested address during the validation step and continue, your application did not save that address and you can submit without an address at all.

## How Can This Be Tested/Reviewed?

Add both a household member w both addresses, and a street2 on a primary applicant on a public application. Move forward and backward steps and ensure the data stays. Ensure the data is on the summary view before submission (only the main address for a household member will show). Ensure after an application is submitted that it shows up correctly in the partners site and that from the detail view you can view a member and from the edit view you can edit a member.

To check the address validation issue, you can enter a valid address and then add some extra characters to the end of the main street field, and then check the second original address box and ensure that if you navigate backward, the data is still filled into the fields and reflects the option you chose, and that if you submit the application the data also comes through.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
